lib/std/os.zig

```diff
@@ -2305,8 +2308,15 @@ pub fn chdir(dir_path: []const u8) ChangeCurDirError!void {
     if (builtin.os.tag == .wasi) {
         @compileError("chdir is not supported in WASI");
     } else if (builtin.os.tag == .windows) {
-        const dir_path_w = try windows.sliceToPrefixedFileW(dir_path);
-        @compileError("TODO implement chdir for Windows");
+        windows.SetCurrentDirectory(dir_path) catch |err| switch (err) {
```

Comment:
The call to `windows.sliceToPrefixedFileW` should stay, and instead of feeding in `[]const u8` directly to `windows.SetCurrentDirectory`, you should change the signature of this function to accept UTF16 encoded paths.

```suggestion
        const dir_path_w = try windows.sliceToPrefixedFileW(dir_path);
        windows.SetCurrentDirectory(dir_path_w.span()) catch |err| switch (err) {
```